examples/jsm/exporters/GLTFExporter.js

```diff
@@ -1102,9 +1102,9 @@ class GLTFWriter {
 
 		}
 
-		if ( options.binary === true ) {
+		let toBlobPromise;
 
-			let toBlobPromise;
+		if ( ( options.binary === true ) || ( canvas.toDataURL === undefined ) ) {
```

Comment:
I'm confused by this check. Is this the most readable way to write this code?

Reply:
I think it is the only way to write it without duplicating the `toBlobPromise` construction.

Reply:
> I'm confused by this check. Is this the most readable way to write this code?

I felt the same when I first read the code. Perhaps more readable code is better for better maintainability even it has redundant code?

Reply:
Is it better like this?

Reply:
Looks much better to me, thanks.

Only one things, it might be better to think of moving the function out of `processImage()` to avoid to make function in every `processImage()` call. But we may do later if needed.

Reply:
Done.